Mage.Sets/src/mage/cards/i/IanMalcolmChaotician.java

```diff
@@ -0,0 +1,117 @@
+package mage.cards.i;
+
+import java.util.UUID;
+import mage.MageInt;
+import mage.MageObject;
+import mage.abilities.Ability;
+import mage.abilities.common.DrawNthCardTriggeredAbility;
+import mage.abilities.condition.common.MyTurnCondition;
+import mage.abilities.effects.OneShotEffect;
+import mage.cards.Card;
+import mage.constants.*;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.players.Player;
+import mage.target.targetpointer.FixedTarget;
+import mage.util.CardUtil;
+import org.apache.log4j.Logger;
+
+/**
+ *
+ * @author jimga150
+ */
+public final class IanMalcolmChaotician extends CardImpl {
+
+    public IanMalcolmChaotician(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{1}{U}{R}");
+        
+        this.supertype.add(SuperType.LEGENDARY);
+        this.subtype.add(SubType.HUMAN);
+        this.subtype.add(SubType.SCIENTIST);
+        this.power = new MageInt(2);
+        this.toughness = new MageInt(2);
+
+        // Whenever a player draws their second card each turn, that player exiles the top card of their library.
+        // During each player's turn, that player may cast a spell from among the cards they don't own exiled with Ian Malcolm, Chaotician, and mana of any type can be spent to cast it.
+        this.addAbility(new IanMalcolmChaoticianAbility());
```

Comment:
This needs to be split into two abilities.

One trigger, and one static allowing each player to cast one card exiled with this Malcolm.

For the trigger, you should accumulate cards in an exileZone for each Malcolm (need to one per MageObjectReference, as Malcolm can blink, which makes you lose playability to all previous exiled cards). Maybe `ExileTargetEffect` with a `setToSourceExileZone` can be of help as reference?

For the static ability, you probably need to have a `Watcher` set up to remember if the player which is the turn played a card or not thanks to Malcolm this turn. The setup for those are not that easy, and this one is even weirder, but maybe [[Johann, Apprentice Sorcerer]] or [[Evelyn, the Covetous]] can help as reference?

There is a lot at work with those kind of effects. Don't hesitate to add unit tests for them.

Reply:
I see why the watcher is necessary, and how the blink behavior needs to work, but i dont think im clear on how exile zone IDs work. If Ian is blinked, wouldn't he get a new UUID, resulting in a different exile zone? What is the relationship between the UUID of a permanent and the MageObjectReference obtained from it?

Reply:
MOR are UUID + zcc.
The UUID of a card doesn't change when that card move zones, which helps for the engine to track movements, but isn't how the rules usually go.

Reply:
Use `CardUtil.getExileZoneId` with offset value (to find stack/battlefield moment -- each move adds +1 to zone change counter). If you exiled cards on stack then find it on battlefield by `CardUtil.getExileZoneId(xxx, -1)`.